### ISSUE TITLE
Mongoose v5 support

### ISSUE THREAD

**Issue by san983** [joegoldbeck/mongoose-encryption#70]
Is there a plan for Mongoose v5 support?  

I'm getting an error at https://github.com/joegoldbeck/mongoose-encryption/blob/master/lib/plugins/mongoose-encryption.js#L264, not sure on how I can help with here. Thanks!

```
TypeError: next is not a function
        at model.<anonymous> (/Users/san983/Projects/Voxbone/voxbone-ai-frontend/node_modules/mongoose-encryption/lib/plugins/mongoose-encryption.js:267:24)
```

**Comment by joegoldbeck** [comment]
Looks like they changed the signature in v5 https://github.com/Automattic/mongoose/blob/master/migrating_to_5.md#init-hook-signatures 

It should be an easy change but I don't have time to fix + test at the moment. If you could create and test a PR that detects whether that `.pre('init'` fn is being passed `(next, doc)` or simply `(doc)`, and only calls next if it exists (obviously), then happy to review and merge fairly quickly :D

**Comment by danielnitu** [comment]
Hi, any updates regarding support for Mongoose v5?

**Comment by arindam89** [comment]
Any updates.

**Comment by zacharybrady** [comment]
Any updates coming?

**Comment by RPallas92** [comment]
any update so far ?

**Comment by yelworc** [comment]
[This fork](https://github.com/colinhemphill/mongoose-encryption) works fine for me, and the changes look clean at first glance – @colinhemphill maybe you could submit those two commits as a PR here?

**Comment by auvos** [comment]
Just checking if anyone is working on this? Or is this plugin becoming obsolete?

**Comment by colinhemphill** [comment]
Problem is that we have a working fix, but the repo owner (understandably) doesn't want to merge until  all of the tests succeed. v5 mongoose breaking the init function here is a simple fix, but tracking down where else it's causing failures is difficult.

As @yelworc said, there's a thorough discussion on PR https://github.com/joegoldbeck/mongoose-encryption/pull/71

This might just be a case where we have to abandon this codebase and write from the ground up for v5+, but making sure it's cross-compatible would be difficult.

**Comment by joegoldbeck** [comment]
Yep, see the conversation in #71. It's pretty close, but there is an issue with the way embedded documents are handled. Once we figure out how to get that test green we should be good to go.

**Comment by joegoldbeck** [comment]
Available as of [2.0.0](https://github.com/joegoldbeck/mongoose-encryption/tree/2.0.0)
